app/models/miq_worker.rb

```diff
@@ -336,7 +336,7 @@ def self.close_pg_sockets_inherited_from_parent
   # This converts it back to a Ruby Array safely.
   def queue_name
     begin
-      JSON.parse(self[:queue_name])
+      JSON.parse(self[:queue_name]).sort
```

Comment:
Will this handle a single queue_name that's not an array?

Reply:
@Fryguy that happens here https://github.com/juliancheal/manageiq/blob/db6a096ebd97d09ab0a58d41c5518e3f69a46d58/app/models/miq_worker.rb#L340-L342 If `JSON` tries to parse a string.